File: spoj/4300-AE00/solution_4300.cpp
#include <iostream>
#include <math.h>

using namespace std;

int factors(int num);
int rectangleCount(int num);

int rectangleCount(int num)
{
    int total = 0;
    for (int i = 2; i <= num; i++)
        total += factors(i);
    return num + total;
}

int factors(int num)
{
        int count = 0;
        for (int i = 2; i <= sqrt(num); i++)
            if (num % i == 0)
                count++;

    return count;
}

int main()
{
    int num;
    cin >> num;
    cout << rectangleCount(num) << "\n";
    return 0;
}

File: spoj/4300-AE00/solution_4300.py
from math import sqrt


def factor_count(num):
    count = 0
    for x in range(2, int(sqrt(num)) + 1):
        if num % x == 0:
            count += 1
    return count


def rectangle_count(num):
    factors = 0
    for x in range(4, num + 1):
        factors += factor_count(x)
    return num + factors


if __name__ == "__main__":
    num = int(input())
    print(rectangle_count(num))
